### PR TITLE
Use v5 roles in RoleFromSpec

### PR DESCRIPTION
**Pull request by nklaassen**
Trying to eliminate our use of V3 roles. This does slightly narrow the system roles used internally, which currently implicitly rely on the v3 role defaults. The differences are:

Label | RoleV3 Default | RoleV5 Default
----- | --------------- | ---------------
`NodeLabels` | `[{"*": "*"}]` if the user has any logins, else `[]` | `nil`
`AppLabels` | `[{"*": "*"}]` | `nil`
`KubernetesLabels` | `[{"*": "*"}]` | `nil`
`DatabaseLabels` | `[{"*": "*"}]` | `nil`

`RoleFromSpec` is only called in `definitionForBuiltinRole` and `authorizeRemoteBuiltinRole`. These roles are not persisted to the backend, and are always generated just-in-time when needed. This should not effect upgrades/downgrades or require any migration.

By removing the V3 defaults, this PR causes the following changes to allow labels for system roles:
Role | Changes
---- | ----
`RoleAuth` |  lost wildcard `NodeLabels`, `AppLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleProvisionToken` | lost wildcard `NodeLabels`, `AppLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleNode` | lost wildcard `AppLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleApp` | lost wildcard `NodeLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleDatabase` | lost wildcard `NodeLabels`, `AppLabels`, and `KubernetesLabels`
`RoleProxy` | unchanged
`Signup` | lost wildcard `NodeLabels`, `AppLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleAdmin` | unchanged
`RoleNop` | lost wildcard `NodeLabels`, `AppLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleKube` | lost wildcard `NodeLabels`, `AppLabels`, and `DatabaseLabels`
`RoleWindowsDesktop` | lost wildcard `NodeLabels`, `AppLabels`, `KubernetesLabels`, and `DatabaseLabels`
`RoleRemoteProxy` | unchanged

Note: `RoleRemoteProxy` appears to need wildcard labels for Apps and Databases, or else tests fail with logs of access being denied to that role. The allow `rules` on the role also include `KindNode` and `KindKubeService` so I've explicitly given `RoleRemoteProxy` all the wildcard labels it currently has from the v3 defaults.

Other system roles require the wildcard labels for their respective type. `RoleAdmin` and `RoleProxy` get wildcard allow for all.

One step towards https://github.com/gravitational/teleport/issues/14345

### CODE REVIEW

File: lib/auth/auth_with_roles_test.go [hunk 1]
```diff
@@ -2881,7 +2881,7 @@ func TestListResources_KindKubernetesCluster(t *testing.T) {
 	srv, err := NewTestAuthServer(TestAuthServerConfig{Dir: t.TempDir()})
 	require.NoError(t, err)
 
-	authContext, err := srv.Authorizer.Authorize(context.WithValue(ctx, ContextUser, TestBuiltin(types.RoleProxy).I))
+	authContext, err := srv.Authorizer.Authorize(context.WithValue(ctx, ContextUser, TestBuiltin(types.RoleAdmin).I))
```
Comment (jakule) [inline lib/auth/auth_with_roles_test.go:2884]: I read the PR description, but why is this test failing with "access denied"? Doesn't this suggest a bug somewhere if we need to increase the privileges in order to make that test pass?

File: lib/auth/tls_test.go [hunk 2]
```diff
@@ -1080,7 +1080,7 @@ func TestAppServerCRUD(t *testing.T) {
 	ctx := context.Background()
 	tt := setupAuthContext(ctx, t)
 
-	clt, err := tt.server.NewClient(TestBuiltin(types.RoleApp))
+	clt, err := tt.server.NewClient(TestBuiltin(types.RoleAdmin))
```
Comment (jakule) [inline lib/auth/tls_test.go:1083]: The same as above. Why updating the role version require to elevate the privilege here?

Comment (nklaassen) [reply lib/auth/tls_test.go]: I changed this test before giving `RoleApp` explicit wildcard allow `AppLabels`, which was a mistake. I've now reverted the change in this test.

**Review by jakule — APPROVED** [review]

**Comment by zmb3** [comment]
Were the `editor`, `access`, and `auditor`, presets already v5?

**Comment by nklaassen** [comment]
> Were the `editor`, `access`, and `auditor`, presets already v5?

yes https://github.com/gravitational/teleport/blame/13d68af6f4161b372578db4e2a172b7cfe0213a7/lib/services/presets.go#L34

**Review by ryanclark — APPROVED** [review]

**Pull request merged**